lib/postgrex/extensions/tsvector.ex

```diff
@@ -0,0 +1,127 @@
+defmodule Postgrex.Extensions.Tsvector do
+  @moduledoc false
+
+  import Postgrex.BinaryUtils, warn: false
+  use Bitwise
+  use Postgrex.BinaryExtension, [send: "tsvectorsend"]
+
+  def encode(_) do
+    quote location: :keep do
+      values when is_list(values) ->
+        encoded_tsvector = unquote(__MODULE__).encode_tsvector(values)
+        <<byte_size(encoded_tsvector) :: int32, encoded_tsvector :: binary>>
+      other ->
+        raise ArgumentError,
+        Postgrex.Utils.encode_msg(other, "a list of tsvectors")
+    end
+  end
+
+  def decode(_) do
+    quote do
+      <<len :: int32, value :: binary-size(len)>> ->
+        <<nb_lexemes :: int32, words :: binary>> = value
+        unquote(__MODULE__).decode_tsvector_values(:binary.bin_to_list(words))
+    end
+  end
+
+  ## Helpers
+
+  def encode_tsvector(values) do
+    << length(values) :: int32, encode_lexemes(values) :: binary >>
+  end
+
+  defp encode_lexemes(values) do
+    Enum.map(values, fn(x) ->
+        case x do
```

Comment:
Indeed! Didn't knew I could pattern match at the function level. Really cool feature.